### ISSUE TITLE
PCem-MacOSX Metal release

### ISSUE THREAD

**Issue by mortinus** [kyr0/PCemV14MacOSX#4]
Hi,
I would like to know if PCem-MacOSX will run on new MAC OS with Metal libraries without the support for OpenGL.

Next fall, with the new Catalina OS, only structured applications for Metal/64bit will be able to run superbly, the other are deprecated.

Bye

**Comment by kyr0** [comment]
Hi mortinus,

yeah, I did ran across that issue as well. This mostly depends on SDL2's Metal support and how compatible the shader interfaces will be. If SDL2 adds support for Metal 3D (2D is supported since 2.0.8) it should be quite easy to update to a new minor/patch version of SDL2 and add Metal to 
https://github.com/kyr0/PCemV14MacOSX/blob/4b3abd05a3852fded1ac77340e7b50c8fa788db5/src/wx-sdl2-video.c#L34

Please note that this repo is not official. There is no official support for MacOS for PCemV14 atm.

Best, 
Aron

**Comment by Haroldo64** [comment]
Great emulator, we hope it will be updated on Mac OS next fall ...and after !!!

Thanks
BYE

**Comment by DaitarnIII** [comment]
Oh yes, this kind of emulartor is a need for Mac OS users !

Thanks

**Comment by UgoCaneFifone** [comment]
Thanks by all Mac OS Users! Great projet !

**Comment by puleyo** [comment]
Wow !
Unofficial mmmm....

This is a great emulator, this must become an official port !
Go on !
Thank you !!!

**Comment by kyr0** [comment]
Thanks a lot to you all! :) I'm happy as well. This project was/is fun and a good opportunity to learn for me. And it's just amazing to play all these games of my childhood on my Macbook in native speed and feeling :)

If you'd like to help this port to gain a little more momentum, you could also show your interest for an official port in this post of the official forum: https://pcem-emulator.co.uk/phpBB3/viewtopic.php?f=4&t=3223&p=12524 

The heavy lifting for PCem has been done by Sarah Walker over the course of more than a decade :)
I believe that official support could be in sight when there is enough community demand and support for MacOS. But it's absolutely up to her - her workload is already "crazy" :)) 

But I'm willing to continue to port upcoming PCem versions as well when there is enough interest (V14 works great for me).

PCemV15 has been released shortly after I released this port. I'm actually thinking about porting V15 to MacOS as well.

Are you interested in the V15 updates? 
- New machines added - Zenith Data SupersPort, Bull Micral 45, Tulip AT Compact, Amstrad PPC512/640, Packard Bell PB410A, ASUS P/I-P55TVP4, ASUS P/I-P55T2P4, Epox P55-VA, FIC VA-503+
- New graphics cards added - Image Manager 1024, Sigma Designs Color 400, Trigem Korean VGA
- Added emulation of AMD K6 family and IDT Winchip 2
- New CPU recompiler. This provides several optimisations, and the new design allows for greater portability and more scope for optimisation in the future
- Experimental ARM and ARM64 host support
- Read-only cassette emulation for IBM PC and PCjr
- Numerous bug fixes


Thanks and best,
Aron

**Comment by Alessandro1970** [comment]
Hi,
Apple has deprecated opengl and next year rosetta2 will begin to abandon us.
It would be necessary to start supporting Metal and AppleSilicon M1.
For the rest PCem will have a great future, I believe!
bye

**Comment by kyr0** [comment]
Hey @Alessandro1970 
actually, as far as I've read, Apple is having support for OpenGL on Apple Silicon: https://developer.apple.com/forums/thread/650427

Do you have more recent info about rosetta removing support for OpenGL completely?
Best :)

**Comment by Alessandro1970** [comment]
Ok, but Metal and Arm code give best result for us...

**Comment by kyr0** [comment]
Yeah, at some point we need to do this :) https://www.raywenderlich.com/9211-moving-from-opengl-to-metal

**Comment by kyr0** [comment]
Another opportunity to learn... xD

**Comment by Alessandro1970** [comment]
Will Pcem have an Universal core ? (intel+ARM)
